presto-main/src/test/java/io/prestosql/sql/planner/assertions/SemiJoinMatcher.java

```diff
@@ -64,6 +77,31 @@ public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session ses
             return NO_MATCH;
         }
 
+        if (hasDynamicFilter.isPresent()) {
+            if (hasDynamicFilter.get()) {
+                if (semiJoinNode.getDynamicFilterId().isEmpty()) {
+                    return NO_MATCH;
+                }
+                DynamicFilterId dynamicFilterId = semiJoinNode.getDynamicFilterId().get();
+                boolean dynamicFilterMatched = searchFrom(semiJoinNode.getSource())
+                        .where(FilterNode.class::isInstance)
+                        .findAll()
+                        .stream()
+                        .flatMap(filterNode -> extractExpressions(filterNode).stream())
+                        .flatMap(expression -> extractDynamicFilters(expression).getDynamicConjuncts().stream())
+                        .filter(descriptor -> descriptor.getId().equals(dynamicFilterId))
+                        .map(descriptor -> Symbol.from(descriptor.getInput()))
+                        .allMatch(sourceSymbol -> symbolAliases.get(sourceSymbolAlias).equals(sourceSymbol.toSymbolReference()));
```

Comment:
you should make sure that there is at least one match (`allMatch` returns true on empty stream), e.g:
```
List<DynamicFilter.Descriptor> matchingDescriptors = searchFrom(semiJoinNode.getSource())
  .where(FilterNode.class::isInstance)
  .findAll()
  .stream()
  .flatMap(filterNode -> extractExpressions(filterNode).stream())
  .flatMap(expression -> extractDynamicFilters(expression).getDynamicConjuncts().stream())
  .filter(descriptor -> descriptor.getId().equals(dynamicFilterId))
  .collect(toImmutableList();
boolean sourceSymbolsMatch = matchingDescriptors.stream()
  .map(descriptor -> Symbol.from(descriptor.getInput()))
  .allMatch(sourceSymbol -> symbolAliases.get(sourceSymbolAlias).equals(sourceSymbol.toSymbolReference()));
if (!matchingDescriptors.isEmpty() && sourceSymbolsMatch) {
  ...
}
```

Reply:
oh good catch. Thanks for the pointer